bitcoin/src/blockdata/transaction.rs

```diff
@@ -718,7 +718,7 @@ impl Transaction {
     pub fn stripped_size(&self) -> Weight {
         let mut input_size: Weight = Weight::ZERO;
         for input in &self.input {
-            input_size += TxIn::BASE_WEIGHT
+            input_size += Weight::from_wu(32 + 4 + 4)
```

Comment:
In 5f8b7cb7e5f48980782ba5e001c3ea6451d869c2:

I'm confused. This seems like it just duplicates the bug and prevents it from being fixed in your later commit that changes `BASE_WEIGHT`.

Reply:
`TxIn::BASE_WEIGHT` should be multiplied by 4.  I also looked at the BDK which re-implements this: https://github.com/bitcoindevkit/bdk/blob/master/crates/bdk/src/wallet/coin_selection.rs#L118.  I guess this implies that prev_txid, prev_vout (4 bytes), sequence are all part of the non-segwit count.  L:721 should not be multiplied by 4, so if I was to leave BASE_WEIGHT as is, then all surrounding tests fail.

Reply:
Something that does confuse me about stripped_size() and scaled_size() is that these number should not be `Weight::from_wu(32 + 4 + 4)` should be multiplied by 4 here since they are not part of the witness portion of the transaction, although the tests fail otherwise.

Reply:
A weight should never be multiplied by 4.

With all due respect, I don't have the time or energy to continue this discussion.

Reply:
Oops sorry.  I appreciate all the feedback so far.

Reply:
Oops sorry.  I appreciate all the feedback so far.

Reply:
Ugh sorry for all the messages.  My internet connection is not great here.

Reply:
> A weight should never be multiplied by 4.

Right, since a wu is already discounted by a factor of four.  My mistake.